lib/install/management/appliance.go

```diff
@@ -190,33 +191,39 @@ func (d *Dispatcher) deleteVM(vm *vm.VirtualMachine, force bool) error {
 		}
 	}
 
-	//removes the vm from vsphere, but detaches the disks first
-	_, err = vm.WaitForResult(d.op, func(ctx context.Context) (tasks.Task, error) {
-		return vm.DeleteExceptDisks(ctx)
-	})
-	// We are getting ConcurrentAccess errors from DeleteExceptDisks - even though we don't set ChangeVersion in that path
-	// We are ignoring the error because in reality the operation finishes successfully.
-	ignore := false
-	if err != nil {
-		if f, ok := err.(types.HasFault); ok {
-			switch f.Fault().(type) {
-			case *types.ConcurrentAccess:
-				d.op.Warn("DeleteExceptDisks failed with ConcurrentAccess error. Ignoring it.")
-				ignore = true
-			}
+	operation := func() error {
+		_, err := vm.DeleteExceptDisks(d.op)
+
+		if err != nil {
```

Comment:
Unclear.

If this is retrying the `vm.Unregister` then the `vm.DeleteExceptDisks` should be outside of the `operation` function.
If this is retrying `vm.DeleteExceptDisks` then the error check should be after the `retry.Do` or we will always unregister even for errors which trigger a retry.

Reply:
You're right, if `DeleteExceptDisks` failed with `ConcurrentAccess` we should retry this separately before unregister as well. Fixing it.